packages/hardhat-ethers/test/no-account.test.ts

```diff
@@ -0,0 +1,129 @@
+import { assert } from "chai";
+import { ethers } from "ethers";
+import { TASK_COMPILE } from "hardhat/builtin-tasks/task-names";
+import { NomicLabsHardhatPluginError } from "hardhat/plugins";
+import { Artifact, HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { SignerWithAddress } from "../src/signers";
+
+import { useEnvironment } from "./helpers";
+
+describe("hardhat-ethers plugin", function () {
+  describe("hardhat network with no accounts", function () {
+    useEnvironment("hardhat-project-no-accounts", "hardhat");
+
+    describe("fixture setup", function () {
+      it("should not have accounts", async function () {
+        const signers = await this.env.ethers.getSigners();
+        assert.isEmpty(signers);
+      });
+    });
+
+    describe("getContractAt", function () {
+      let signerAddress: string;
+
+      beforeEach(async function () {
+        await this.env.network.provider.request({
+          method: "evm_mine",
+          params: [],
+        });
+
+        const { miner } = await this.env.ethers.provider.getBlock("latest");
+        signerAddress = miner;
+
+        await this.env.run(TASK_COMPILE, { quiet: true });
+      });
+
+      describe("with the name and address", function () {
+        it("Should return an instance of a contract with a read-only provider", async function () {
+          const receipt = await deployGreeter(this.env, signerAddress);
+
+          assert.isTrue(this.env.ethers.provider._isProvider);
+          const contract = await this.env.ethers.getContractAt(
+            "Greeter",
+            receipt.contractAddress
+          );
+
+          assert.isDefined(contract.provider);
+          assert.isNotNull(contract.provider);
+          assert.strictEqual(contract.provider, this.env.ethers.provider);
```

Comment:
I'd consider this strict equality an implementation detail. It could be a clone.